MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -18,16 +18,25 @@ final class FeedViewModel {
 		Localized.Feed.title
 	}
 	
+	var errorMessage: String {
+		Localized.Feed.loadError
+	}
+	
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onFeedLoadFailed: Observer<String>?
```

`onFeedLoadFailed: Observer<String> ` could be `onErrorStateChange: Observer<String?>`, so the ViewModel decides which message to show and when to show/hide errors (instead of the view controller). It's often easier to manage state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed. And when there's an error: `onErrorStateChange(Localized.Feed.errorMessage)`.

---

This is a very good point, subtle difference, but important. I have refactored as you suggested.